cmake/UnitTests.cmake

```diff
@@ -391,6 +391,26 @@ list(APPEND CPU_OPENCL_FAILED_TESTS "Unit_hipMallocPitch_KernelLaunch - float")
 list(APPEND CPU_OPENCL_FAILED_TESTS "Unit_hipMallocPitch_KernelLaunch - double") # Failed
 list(APPEND CPU_OPENCL_FAILED_TESTS "Unit_hipGraphAddHostNode_ClonedGraphwithHostNode") # Failed
 
+# broken tests, they all try to write outside allocated memory;
+# valgrind + pocl shows:
```

Comment:
Should be added to POCL exclusion tests. 
Also, these should be reproducible on the CI system.

Reply:
You mean they cannot be reproduced in the CI system? I think if it fails for someone we should still mark it (flaky?), otherwise we keep having uncertainty which tests are expected to work or not.

Reply:
It does not seem like the CI pulls in any memory error detection analyzer/instrumentation so it's just luck that the out-of-bound accesses do not cause any troubles.

Reply:
> I think if it fails for someone we should still mark it (flaky?), otherwise we keep having uncertainty which tests are expected to work or not.

I think we  should investigate why it fails for you and not for CI, especially for PoCL where we know that it's not related to the GPU driver.

Reply:
> It does not seem like the CI pulls in any memory error detection analyzer/instrumentation so it's just luck that the out-of-bound accesses do not cause any troubles.

Any idea on what to do so that the CI system behavior matches yours?

Reply:
I understood from the above that if you run it under valgrind, it should reveal writes that write over the buffers, but of course these type of errors might get unnoticed if they do not happen to overwrite anything essential. This is a major reason why we port to PoCL-CPU, to make it easier to debug issues like this.